packages/shared/src/__tests__/keys.test.ts

```diff
@@ -164,3 +165,19 @@ describe('isProductionFromSecretKey(key)', () => {
     expect(result).toEqual(expected);
   });
 });
+
+describe('getCookieSuffix(publishableKey)', () => {
+  const cases: Array<[string, string]> = [
+    ['pk_live_Y2xlcmsuY2xlcmsuZGV2JA', '1Z8AzTQD'],
+    ['pk_test_Y2xlcmsuY2xlcmsuZGV2JA', 'QvfNY2dr'],
+  ];
+
+  test.each(cases)('given %p pk, returns %p cookie suffix', async (pk, expected) => {
+    expect(await getCookieSuffix(pk)).toEqual(expected);
+  });
+
+  test('omits special characters from the cookie suffix', async () => {
+    const pk = 'pk_test_ZW5vdWdoLWFscGFjYS04Mi5jbGVyay5hY2NvdW50cy5sY2xjbGVyay5jb20k';
+    expect(await getCookieSuffix(pk)).toEqual('jtYvytHq');
```

Comment:
Is there any scenario where we can end up with a suffix whose length is less than 8 chars? For example, if there's an edge case where we need to remove many special characters? 

I'm almost sure that there's no such case, but asking just to be safe.

Reply:
For such case to exist, we would have to create a hash digest with more than 32 (out of 40) bytes with a value that will be encoded to special char.
I will search for the sha1 RFC and try to identify if and when would that be possible

Reply:
Great 💯 
Resolving this one as it shouldn't block you. Its very unlikely that this could happen, but if you find anything just add another comment here.